target/linux/ramips/dts/mt7621_cudy-wr2100.dts

```diff
@@ -0,0 +1,244 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "cudy,wr2100", "mediatek,mt7621-soc";
+	model = "Cudy WR2100";
+
+	aliases {
+		led-boot = &led_blue;
+		led-failsafe = &led_blue;
+		led-running = &led_blue;
+		led-upgrade = &led_blue;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_blue: blue {
+			label = "blue";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+		};
+
+		red {
+			label = "red";
+			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+		};
+
+		wan {
+			label = "green:wan";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		lan1 {
+			label = "green:lan1";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		lan2 {
+			label = "green:lan2";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+		};
+
+		lan3 {
+			label = "green:lan3";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+
+		lan4 {
+			label = "green:lan4";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 10 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+&gmac0 {
+	// port 6 on the switch
+	mtd-mac-address = <&bdinfo 0xde00>;
+	phy-mode = "trgmii"; // TODO overrides rgmii
+};
+
+// &gmac1 {
+// 	status = "okay";
+// 	// port 5 on the switch
+//
+// 	mtd-mac-address = <&bdinfo 0xde00>;
+// 	phy-mode = "rgmii"; // TODO overrides rgmii-rxid
+//
+// 	fixed-link {
+// 		speed = <1000>;
+// 		full-duplex;
+// 		pause;
+// 	};
+// };
+
+// &hnat {
+// 	mtketh-wan = "eth1";
+// 	mtketh-ppd = "eth0";
+// 	ext-devices = "rai0", "ra0", "rai1", "ra1", "apclii0", "apcli0";
+// 	mtketh-max-gmac = <2>;
+// 	status = "okay";
+// };
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		//device_type = "pci";
```

Comment:
```device-type = "pci"``` is specified in the root of pcie in *mt7621.dtsi* and it's a property for PCI host controller. So this can be dropped. (same for line 127)

Reply:
i assumed as much.
but because there are still difficulties with the wifi adapters, i kept those lines around.
with the confirmation, that those lines don't have any effect, i'm dropping them